shelley-ma/impl/src/Cardano/Ledger/Allegra/Translation.hs

```diff
@@ -3,8 +3,16 @@
 {-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE MultiParamTypeClasses #-}
 {-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
 {-# LANGUAGE TypeFamilies #-}
 {-# LANGUAGE UndecidableInstances #-}
+{- We disable warnings for name shadowing because of
+https://gitlab.haskell.org/ghc/ghc/-/issues/14630, which means that we get
+shadowing warnings for the named field puns when used with a pattern synonym.
+-}
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
```

Comment:
Try the `DisambiguateRecordFields` extension, that often helps in these situations.

Reply:
But not this one, alas